### ISSUE TITLE
WGAN-GP gradient penalty not backpropagating into discriminator weights (createGraph=false in nested tape)

### ISSUE THREAD

**Issue by ooples** [ooples/AiDotNet#1844]
## Bug

The WGAN-GP gradient penalty across every synthetic-data GAN file computes the penalty VALUE but does not backpropagate the penalty gradient into the discriminator's weights. The critic trains against the Wasserstein term alone, not the intended \`Wasserstein + λ·GP\` objective.

## Affected files
- \`src/NeuralNetworks/SyntheticData/CTGANGenerator.cs\` \`ComputeGradientPenalty\` @ line 949
- \`src/NeuralNetworks/SyntheticData/CausalGANGenerator.cs\` \`ComputeGradientPenalty\` @ line 1043
- \`src/NeuralNetworks/SyntheticData/CopulaGANGenerator.cs\` \`ComputeGradientPenalty\` @ line 875
- \`src/NeuralNetworks/SyntheticData/DPCTGANGenerator.cs\` \`ComputeGradientPenalty\` @ line 873
- \`src/NeuralNetworks/SyntheticData/TableGANGenerator.cs\` \`ComputeGradientPenalty\` @ line 494

## Root cause

Each ComputeGradientPenalty implementation runs the critic forward inside an **inner** GradientTape and computes gradients of scores wrt interpolated inputs:

\`\`\`csharp
using (var gradientTape = new GradientTape<T>())
{
    var scores = DiscriminatorForwardBatched(interpolated, true);
    var summedScores = Engine.ReduceSum(scores, scoreAxes, keepDims: false);
    var gradients = gradientTape.ComputeGradients(summedScores, [interpolated]);  // ← createGraph defaults to FALSE
    inputGradients = gradients[interpolated];
}
\`\`\`

Because \`ComputeGradients\` defaults to \`createGraph=false\`, the inner backward's ops are NOT recorded on the outer tape. \`inputGradients\` therefore has \`GradFn = null\` from the outer tape's perspective — the outer tape sees it as a leaf tensor. The subsequent penalty computation (\`||∇||₂² - 1)²\`) is on-tape but its dependency chain terminates at \`inputGradients\` with no path back to the discriminator weights.

Result: the critic's gradient update only reflects \`-E[D(fake)] + E[D(real)]\`, not the \`+ λ·GP\` term. The 1-Lipschitz constraint from Gulrajani et al. 2017 is not actually enforced. WGAN-GP silently degrades to plain WGAN — which needs weight clipping to be stable and diverges without it.

## Fix

Add \`createGraph: true\` to the inner \`ComputeGradients\` call:

\`\`\`csharp
var gradients = gradientTape.ComputeGradients(summedScores, [interpolated], createGraph: true);
\`\`\`

This records the inner backward's ops on the outer tape so \`inputGradients\` has a valid GradFn chain back through the discriminator forward, and the outer tape can differentiate the penalty through the disc weights.

## Discovered while
Working on PR #1843 (GPU-resident fused training for non-time-series models). Was trying to wire the WGAN-GP disc steps through the compiled fused plan; the analysis revealed the disc step wasn't actually training on the GP term even on the eager path.

## Test plan
- Baseline: before fix, disc weights should be unchanged when only \`+ λ·GP\` is added to a baseline \`-E[D(fake)] + E[D(real)]\` loss (proves current implementation ignores the GP gradient).
- Post-fix: disc weights should change measurably when the GP weight is increased (proves the penalty gradient flows into disc weights).
- End-to-end: WGAN-GP disc scores should stay within the 1-Lipschitz constraint band across training (Gulrajani et al. 2017 §5).

## Impact
Correctness — affects the training dynamics of every WGAN-GP-based synthetic data generator. Not a security/data issue, but the models likely train sub-optimally today.
